src/overlays/actors/ovl_En_Mb/z_en_mb.c

```diff
@@ -21,99 +19,1454 @@ const ActorInit En_Mb_InitVars = {
+        if (this->unk_32E != 0) {
+            SkelAnime_ChangeAnimDefaultStop(&this->skelAnime, &D_0600E18C);
+            this->unk_32E = 0;
+            func_800AA000(this->actor.xzDistFromLink, 0xFF, 0x14, 0x96);
+            func_8005AA1C(&globalCtx->mainCamera, 2, 25, 5);
+        } else {
+            func_80AA6898(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mb/func_80AA840C.s")
+void func_80AA840C(EnMb* this, GlobalContext* globalCtx) {
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mb/func_80AA8514.s")
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+        if (this->unk_32E != 0) {
+            this->unk_32E = (s16)(this->unk_32E - 1);
+            if (this->unk_32E == 0) {
+                if (this->unk_32A == 0) {
+                    SkelAnime_ChangeAnim(&this->skelAnime, &D_0600E18C, 3.0f, 0.0f,
+                                         SkelAnime_GetFrameCount(&D_0600E18C), 3, 0.0f);
+                    this->unk_32A = 1;
+                    this->unk_32E = 6;
+                } else {
+                    SkelAnime_ChangeAnim(&this->skelAnime, &D_0600E18C, 3.0f, 0.0f,
+                                         SkelAnime_GetFrameCount(&D_0600E18C), 3, 0.0f);
+                }
+            }
+        } else {
+            func_80AA6898(this);
+        }
+    }
+}
 
+void func_80AA8514(EnMb* this, GlobalContext* globalCtx) {
+    static Vec3f D_80AA9D78 = { 0.0f, 0.0f, 0.0f };
+    Vec3f effPosition;
+    Vec3f temp;
+    Vec3f effZeroVector;
+    s32 effectsPerFrame;
+
+    effPosition = this->actor.posRot.pos;
+    effPosition.x += Math_Sins(this->actor.shape.rot.y) * -70.0f;
+    effPosition.z += Math_Coss(this->actor.shape.rot.y) * -70.0f;
+    Math_SmoothScaleMaxMinF(&this->actor.speedXZ, 0.0f, 1.0f, 0.5f, 0.0f);
+    temp = effPosition;
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+        if (this->unk_32A > 0) {
+            effZeroVector = D_80AA9D78;
+            this->unk_32A--;
+            for (effectsPerFrame = 4; effectsPerFrame >= 0; effectsPerFrame--) {
+                effPosition.x = Math_Rand_CenteredFloat(240.0f) + temp.x;
+                effPosition.y = Math_Rand_CenteredFloat(15.0f) + (temp.y + 20.0f);
+                effPosition.z = Math_Rand_CenteredFloat(240.0f) + temp.z;
+                EffectSsDeadDb_Spawn(globalCtx, &effPosition, &effZeroVector, &effZeroVector, 0xE6, 7, 0xFF, 0xFF, 0xFF,
+                                     0xFF, 0, 0xFF, 0, 1, 9, 1);
+            }
+            return;
+        } else {
+            Item_DropCollectibleRandom(globalCtx, &this->actor, &effPosition, 0xC0);
+            Actor_Kill(&this->actor);
+            return;
+        }
+    }
+    if (((s32)this->skelAnime.animCurrentFrame == 15) || ((s32)this->skelAnime.animCurrentFrame == 22)) {
+        func_800AA000(this->actor.xzDistFromLink, 0xFF, 0x14, 0x96);
```

Comment:
if you make the frame 15 || 22 case an `else if` of the above if, you can remove the early returns